texastribune/donations
# Move form validation to store level

#### What's this PR do?
Moves the data for whether a particular form input is valid to the store/Vuex level (as opposed to the local component level).

#### Why are we doing this? How does it help us?
To stay consistent. I should have done it this way from the beginning. If we're storing an input's actual value at the Vuex level, it just makes sense to also have its validity in the same data store. Doing this will also make it much easier for us to put multiple forms on a page, which I imagine could come up in UMP.

This also gave me a chance to clean up a few code smells, again in preparation for UMP where I suspect this app's JS will get quite the workout.

#### How should this be manually tested?
+ `make`
+ `make restart`
+ Go to `/donate`. Ensure validation for amount, email address, first name, last name and ZIP works. Ensure you can submit successfully.
+ Go to `/circle`. Ensure validation for email address, first name, last name and ZIP works. Ensure you can submit successfully.
+ Go to `/business`. Ensure validation for every field (except state and "encouraged to give by", which don't have validation) works. Ensure you can submit successfully.

#### How should this change be communicated to end users?
NA.

#### Are there any smells or added technical debt to note?
Less. This has been bothering me for a while.

#### What are the relevant tickets?
https://3.basecamp.com/3098728/buckets/736178/todos/1514095590

#### Have you done the following, if applicable:
* [ ] Added automated tests? *(someday)*
